src/common/ina.rs

```diff
@@ -0,0 +1,246 @@
+        max_current_nano_amperes: i64,
+        shunt_resistance_nano_ohms: i64,
+    ) -> anyhow::Result<Self> {
+        let current_reading_lsb = max_current_nano_amperes / (1 << 15);
+        let (unadjusted_calibration_scale, power_reading_lsb) = match model {
+            Model::Ina219 => {
+                let power_lsb = (max_current_nano_amperes * 20 + (1 << 14)) / (1 << 15);
+                (INA_219_CALIBRATION_SCALE, power_lsb)
+            }
+            Model::Ina226 => (INA_226_CALIBRATION_SCALE, 25 * current_reading_lsb),
+        };
+        let calibration_scale = (unadjusted_calibration_scale
+            / ((current_reading_lsb as f64) * (shunt_resistance_nano_ohms as f64 * 1e-9)))
+            as i64;
+        if calibration_scale >= (1 << 16) {
+            anyhow::bail!(
+                "ina calibrate: calibration register value invalid {:?}",
+                calibration_scale
+            );
+        }
+        // let address: [u8; 1] = [0x05];
+        let calibration_scale_bytes = (calibration_scale as u16).to_be_bytes();
+        Ok(Self {
+            model,
+            i2c_handle,
+            i2c_address,
+            calibration_scale_bytes,
+            max_current_nano_amperes,
+            power_reading_lsb,
+        })
+    }
+
+    fn calibrate(&mut self) -> anyhow::Result<()> {
+        // set scaling factor for current and power registers by writing adjusted
+        // calibration scale to the appropriate register
+        let mut calibration_scale_bytes: [u8; 2] = [0; 2];
+        calibration_scale_bytes.copy_from_slice(&self.calibration_scale_bytes);
+        self.write_to_register(0x05, &calibration_scale_bytes)?;
+        // reset the config register to default values
+        let default_config_register_bytes = DEFAULT_CONFIG_REGISTER_VALUE.to_be_bytes();
+        self.write_to_register(CONFIG_REGISTER, &default_config_register_bytes)
+    }
+
+    fn write_to_register(&mut self, register: u8, buffer: &[u8]) -> anyhow::Result<()> {
+        let mut byte_vec: Vec<u8> = vec![register];
+        byte_vec.extend_from_slice(buffer);
+        self.i2c_handle.write_i2c(self.i2c_address, &byte_vec)
+    }
+}
+
+impl<H> PowerSensor for Ina<H>
+where
+    H: I2CHandle,
+{
+    fn get_voltage(&mut self) -> anyhow::Result<Voltage> {
+        let mut voltage_bytes: [u8; 2] = [0; 2];
+        self.i2c_handle
+            .write_read_i2c(self.i2c_address, &VOLTAGE_REGISTER, &mut voltage_bytes)?;
+        let volts = match self.model {
+            Model::Ina226 => (i16::from_be_bytes(voltage_bytes) as f64) * 1.25e-3,
+            Model::Ina219 => ((i16::from_be_bytes(voltage_bytes) >> 3) as f64) / 250.0,
+        };
+        Ok(Voltage {
+            volts,
+            power_supply_type: PowerSupplyType::DC,
+        })
+    }
+
+    fn get_current(&mut self) -> anyhow::Result<Current> {
+        let current_reading_lsb = self.max_current_nano_amperes / (1 << 15);
+        self.calibrate()?;
```

Comment:
Is it necessary to calibrate on every call to `get_current` and `get_power`?

Reply:
potentially not, I cribbed this part from RDK as well. But RDK might have been doing this to accommodate reconfiguration of something like that, so I'll try removing and seeing what happens.